### PR TITLE
Server client

### PR DESCRIPTION
**Pull request by Alban-Bodenes**

### CODE REVIEW

File: Sources/Client/GraphicalLib/GraphicalLib.hpp [hunk 1]
```diff
@@ -20,40 +20,83 @@
...
+            private:
+                sf::RenderWindow *_renderWindow;
                 sf::Mouse _mouse;
                 sf::View _view;
 
                 float _viewX;
+
+                std::map<size_t, std::shared_ptr<Sprite>> _spriteMap; // There is an id Jules
+                std::map<size_t, std::shared_ptr<Text>> _textMap; // There is an id Jules
+                std::map<size_t, std::shared_ptr<Sound>> _soundMap; // There is an id Jules
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/GraphicalLib.hpp]: comment to delete

File: Sources/Client/GraphicalLib/GraphicalLib.hpp [hunk 2]
```diff
@@ -20,40 +20,83 @@
...
+                std::map<size_t, std::shared_ptr<Music>> _musicMap; // There is an id Jules
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/GraphicalLib.hpp]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 3]
```diff
@@ -27,25 +27,68 @@
             public:
                 virtual ~IGraphicalLib() = default;
 
-                virtual void draw() = 0;
-                virtual void play(Sound &sound) = 0;
-                virtual void play(Music &music) = 0;
+                
+            public:
+                virtual void draw(size_t id) = 0; // for sprite
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:32]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 4]
```diff
@@ -27,25 +27,68 @@
...
+                virtual void print(size_t id) = 0; // for text
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:33]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 5]
```diff
@@ -27,25 +27,68 @@
...
+                virtual void start(size_t id) = 0; // for sound
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:34]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 6]
```diff
@@ -27,25 +27,68 @@
...
+                virtual void play(size_t id) = 0; // for music
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:35]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 7]
```diff
@@ -27,25 +27,68 @@
...
+                virtual void stop(size_t id) = 0; //for music
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:36]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp [hunk 8]
```diff
@@ -27,25 +27,68 @@
...
+                virtual void pause(size_t id) = 0; //for music
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:37]: comment to delete

File: Sources/Client/Network/NetworkManager/NetworkManager.hpp [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+        /**
+         * \class NetworkManager NetworkManager.hpp
+         * 
+         * \brief class that contain methode to manage network
+         * 
+         */
+        class NetworkManager : public INetworkManager {
+            public:
+                /**
+                 * \fn explicit NetworkManager()
```
Comment (Jsavang) [inline Sources/Client/Network/NetworkManager/NetworkManager.hpp:26]: to delete

File: Sources/Client/Network/NetworkManager/NetworkManager.hpp [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+                 * 
+                 * \brief ctor networkManager
+                 */
+                explicit NetworkManager() : _context(), _worker(), _tcpClients(), _thread() {};
+
+                /**
+                 * \fn virtual ~NetworkManager()
+                 *
```
Comment (Jsavang) [inline Sources/Client/Network/NetworkManager/NetworkManager.hpp:34]: wrong prototype in doc

File: Sources/Client/Network/NetworkManager/NetworkManager.hpp [hunk 11]
```diff
@@ -0,0 +1,136 @@
...
+                 * \fn virtual ~NetworkManager()
+                 * 
+                 * \brief dtor NetworkManager
+                 * 
+                 */
+                virtual ~NetworkManager() = default;
+
+                /**
+                 * \fn void start()
+                 *
```
Comment (Jsavang) [inline Sources/Client/Network/NetworkManager/NetworkManager.hpp:42]: wrong doc

File: Sources/Client/Network/NetworkManager/NetworkManager.hpp [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+                 * \fn void start()
+                 * 
+                 * \brief start connection
+                 * 
+                 */
+                void start() override;
+
+                /**
+                 * \fn void stop()
+                 *
```
Comment (Jsavang) [inline Sources/Client/Network/NetworkManager/NetworkManager.hpp:50]: wrong doc

File: Sources/Commons/INetwork/ITCPServer.hpp [hunk 13]
```diff
@@ -63,6 +65,19 @@
...
+
+
+            // /**
+            //  * \fn std::shared_ptr<Buffer> getData()
+            //  * 
+            //  * \brief Get the Data object
+            //  * 
+            //  * \return std::shared_ptr<Buffer> 
+            //  */
+            // virtual std::shared_ptr<Buffer> getBuffer() =  0;
```
Comment (MaelTeyssedre) [inline Sources/Commons/INetwork/ITCPServer.hpp:77]: to delete

File: Sources/Server/Network/NetworkManager/NetworkManager.hpp [hunk 14]
```diff
@@ -14,120 +14,122 @@
...
+
+                asio::io_context _context; /*! asio context */
+                asio::io_context::work _worker; /*! asio worker */
+                std::vector<IUDPSocket *> _udpSockets; /*! vector of udp socket */
+                std::vector<ITCPServer *> _tcpServers; /*! vector of tcp server */
+                std::thread _thread; /*! thread  of  network*/
+        };
+    }
 
 #endif /* !NETWORKMANAGER_HPP_ */
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/NetworkManager/NetworkManager.hpp:135]: Add line

File: Sources/Client/Network/Packet/cPacket.cpp [hunk 15]
```diff
@@ -160,7 +160,12 @@ size_t Packet::pack(std::vector<uint8_t> &data)
...
-std::shared_ptr<std::vector<uint8_t>> Packet::unpack()
+std::vector<uint8_t> rtype::Packet::unpack()
 {
     return (_rawData);
+}
+
+void rtype::Packet::setId(size_t id)
+{
+    _id = id;
 }
```
Comment (Jsavang) [inline Sources/Client/Network/Packet/cPacket.cpp:171]: miss a newline

File: Sources/Server/Network/Packet/Packet.cpp [hunk 16]
```diff
@@ -160,7 +160,22 @@ size_t Packet::pack(std::vector<uint8_t> &data)
...
+
+size_t rtype::Packet::getId() const
+{
+    return (_id);
+}
+
+std::vector<uint8_t> rtype::Packet::getData()
 {
     return (_rawData);
 }
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/Packet/Packet.cpp:181]: Add line

File: Sources/Server/Network/Packet/Packet.hpp [hunk 17]
```diff
@@ -10,52 +10,61 @@
...
+
+            private:
+
+                std::vector<uint8_t> _rawData; /*! data */
+                size_t _packetSize; /*! size of the packet */
+                size_t _id;
+        };
+    }
 
 #endif /* !PACKET_HPP_ */
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/Packet/Packet.hpp:70]: Add line

File: Sources/Client/Network/Packet/cPacket.hpp [hunk 18]
```diff
@@ -10,52 +10,57 @@
...
+                * 
+                * \param data to pack
+                * 
+                * \return size readed
+                */
+                size_t pack(std::vector<uint8_t> &data) override;
+
+                /**
+                * \fn std::shared_ptr<std::vector<uint8_t>> unpack()
+                *
```
Comment (Jsavang) [inline Sources/Client/Network/Packet/cPacket.hpp:49]: wrong doc

File: Sources/Server/Network/TCP/tcpServer.cpp [hunk 19]
```diff
@@ -48,7 +48,32 @@ void TCPServer::startAccept()
...
+    for (size_t i = 0; i < _mapUser.size(); i++)
+    {
+     //   rtype::Packet packetPtr;
+        IPacket *packetPtr = new rtype::Packet;
+        packetPtr->pack(_mapUser[i]->getInput());
+        packetPtr->setId(i);
+        _buffers.emplace(packetPtr);
+    }
+    return (_buffers);
 }
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/TCP/tcpServer.cpp:79]: Add line

File: Sources/Client/RtypeClient.cpp [hunk 20]
```diff
@@ -1,75 +1,152 @@
...
 }
 
 /*
  * Components 
-*/
+ */
+
+void rtype::RtypeClient::_setupCurrentSceneComponent()
+{
+    // struct components::currentScene_s scene = {constants::SCENE::LOADING_MENU, false};
```
Comment (MaelTeyssedre) [inline Sources/Client/RtypeClient.cpp:76]: to delete

File: Sources/Server/Network/UDP/udpSocket.cpp [hunk 21]
```diff
@@ -1,16 +1,16 @@
...
 {
-    _socket.async_send_to(asio::buffer(*data.unpack()), _endpoint, std::bind(&UDPSocket::handleSend, this));
+    _socket.async_send_to(asio::buffer(data.unpack()), _endpoint, std::bind(&rtype::UDPSocket::handleSend, this));
 }
 
-void UDPSocket::handleSend()
+void rtype::UDPSocket::handleSend()
 {
-    std::cout << "send data via udp" << std::endl;
+    // std::cout << "send data via udp" << std::endl;
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/UDP/udpSocket.cpp:10]: Comment

File: Sources/Server/Network/UDP/udpSocket.cpp [hunk 22]
```diff
@@ -23,5 +23,5 @@ void UDPSocket::receive()
                                        receive();
                                    }
                                });
-    std::cout << _data << std::endl;
+    // std::cout << _data << std::endl;
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/UDP/udpSocket.cpp:26]: Comment

File: Sources/Client/RtypeClient.cpp [hunk 23]
```diff
@@ -1,75 +1,152 @@
...
+
+void rtype::RtypeClient::_setupUpdateScene()
+{
+    _r.addSystem(_graphSystem, _r.getComponents<components::mouseState_s>(), _r.getComponents<components::keyState_s>(), _r.getComponents<components::currentScene_s>());
+}
+
+bool rtype::RtypeClient::checkStatus()
+{
+    return (_status);
+}
```
Comment (MaelTeyssedre) [inline Sources/Client/RtypeClient.cpp:152]: end of line

File: Sources/Client/Network/TCP/cTCPClient.hpp [hunk 24]
```diff
@@ -15,100 +15,106 @@
...
+                 * 
+                 * \param context asio context
+                 * \param socket asio socket 
+                 * \param host host to connect
+                 * \param port port to connect
+                 */
+                explicit TCPClient(asio::io_context &context, std::shared_ptr<asio::ip::tcp::socket> socket, std::string host, std::string port);
+
+                /**
+                 * \brief Destroy the TCPClient object
```
Comment (Jsavang) [inline Sources/Client/Network/TCP/cTCPClient.hpp:46]: wrong doc

File: Sources/Client/Systems/Systems.hpp [hunk 25]
```diff
@@ -0,0 +1,17 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
+** Systems
+*/
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/Systems.hpp:6]: to delete

File: Sources/Server/Network/User/tcpUser.cpp [hunk 26]
```diff
@@ -53,7 +51,13 @@ void tcpUser::doWrite(const std::error_code &ec, std::size_t bytes_transfered)
...
+std::vector<uint8_t> &rtype::tcpUser::getInput()
 {
     return (_input);
 }
+
+
+size_t &rtype::tcpUser::getSizeInput()
+{
+    return (_sizeInput);
+}
```
Comment (Alban-Bodenes) [inline Sources/Server/Network/User/tcpUser.cpp:63]: Add line

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp [hunk 27]
```diff
@@ -0,0 +1,58 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
+** UpdateDirection
+*/
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp:6]: to delete

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp [hunk 28]
```diff
@@ -0,0 +1,58 @@
...
+
+#include "UpdateDirection.hpp"
+
+// TODO rajouter le composant controllable
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp:10]: done

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp [hunk 29]
```diff
@@ -0,0 +1,58 @@
...
+            dir.value().x = -1;
+            dir.value().y = -1;
+        }
+        if ((key.value().keyUp && key.value().keyDown) || (key.value().keyLeft && key.value().keyRight) || (key.value().keyDown && key.value().keyLeft && key.value().keyRight && key.value().keyUp)) {
+            dir.value().x = 0;
+            dir.value().y = 0;
+        }
+    }
+
+}
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp:58]: end of line

File: Sources/Server/PacketManager/PacketManager.cpp [hunk 30]
```diff
@@ -1,45 +1,61 @@
...
+
+void rtype::PacketManager::setBuffer(std::shared_ptr<Buffer> buffer)
+{
+    (void)buffer;
 }
+
+std::vector<rtype::Packet> rtype::PacketManager::getRequests()
+{
+    return _packets;
+}
```
Comment (Alban-Bodenes) [inline Sources/Server/PacketManager/PacketManager.cpp:61]: Add line

File: Sources/Client/Network/TCP/cTCPClient.hpp [hunk 31]
```diff
@@ -15,100 +15,106 @@
...
+                 * 
+                 * \brief handle write
+                 * 
+                 * \param ec error code returned
+                 * \param bytes numbers of bytes write
+                 */
+                void doWrite(const std::error_code &ec, size_t bytes);
+
+                /**
+                 * \fn std::shared_ptr<Buffer> getData()
```
Comment (Jsavang) [inline Sources/Client/Network/TCP/cTCPClient.hpp:96]: wrong doc

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp [hunk 32]
```diff
@@ -0,0 +1,23 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
+** UpdateDirection
+*/
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp:6]: to delete

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp [hunk 33]
```diff
@@ -0,0 +1,66 @@
...
+            private:
+                void _updateEvent(SparseArray<components::mouseState_s> &mouseStates, SparseArray<components::keyState_s> &keyStates);
+                void _updateGraph(Registry &r, SparseArray<components::mouseState_s> &mouseStates, SparseArray<components::keyState_s> &keyStates, SparseArray<components::currentScene_s> &currentScenes);
+
+            private:
+                void _setupScenes();
+                void _setupExecScene();
+                void _setupDeleteScene();
+
+            private: // * fill _setupScene map
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:32]: delete comment

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp [hunk 34]
```diff
@@ -0,0 +1,66 @@
...
+                void _setupLoadingMenuScene();
+                void _setupLoadingGameScene() {};
+                void _setupGameScene() {};
+                void _setupParamsScene() {};
+                void _setupMainMenuScene();
+                
+            private: // * fill _execScene map
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:39]: delete comment

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp [hunk 35]
```diff
@@ -0,0 +1,66 @@
...
+                void _setupExecLoadingMenuScene();
+                void _setupExecLoadingGameScene() {};
+                void _setupExecGameScene() {};
+                void _setupExecParamsScene() {};
+                void _setupExecMainMenuScene();
+                
+            private: // * fill _deleteScene map
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:46]: delete comment

File: Sources/Client/Network/UDP/cUDPSocket.hpp [hunk 36]
```diff
@@ -6,82 +6,83 @@
...
-             * \brief send data
-             * 
-             * \param data data to send
-             * 
-             */ 
-            void send(IPacket &data) override;
+              /**
+               * \fn explicit UDPSocket(asio::io_context &context, std::uint16_t port)
+               * 
+               * \brief ctor udp client
```
Comment (Jsavang) [inline Sources/Client/Network/UDP/cUDPSocket.hpp:28]: wrong doc

File: Sources/Client/Network/UDP/cUDPSocket.hpp [hunk 37]
```diff
@@ -6,82 +6,83 @@
...
-             * \fn  void handleSend()
-             * 
-             * \brief handle after receive
-             *
-             */ 
-            void handleSend();
+              /**
+               * \fn std::vector<uint8_t> &receive()
+               * 
+               * \brief receive data from network
```
Comment (Jsavang) [inline Sources/Client/Network/UDP/cUDPSocket.hpp:63]: wrong doc

File: Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp [hunk 38]
```diff
@@ -0,0 +1,130 @@
...
+        break;
+    default:
+        break;
+    }
+    if (!network.value().sendRequest.empty()) {
+        IPacket *packet = new rtype::Packet;
+        (*packet).pack(network.value().sendRequest.front());
+        _tcpClient->send(*packet);
+    }
+}
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp:130]: end of line

File: Sources/Server/Systems/UpdateNetwork.cpp [hunk 39]
```diff
@@ -0,0 +1,59 @@
...
+        break;
+    case 16:
+        if (network.value().request16.empty())
+            network.value().request16.push_back(std::vector<uint8_t>());
+        network.value().request16.back().push_back(opCode);
+        break;
+    case 17:
+        buffer->readFromBuffer(8, reply);
+        if (network.value().request17.empty())
+            network.value().request17.push_back(std::vector<uint8_t>());
```
Comment (Alban-Bodenes) [inline Sources/Server/Systems/UpdateNetwork.cpp:42]: Comment

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp [hunk 40]
```diff
@@ -0,0 +1,22 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
+** UpdatePosition
+*/
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp:6]: to delete

File: Sources/Client/RtypeClient.cpp (hunk 20 above)
Comment (Jsavang) [inline Sources/Client/RtypeClient.cpp]: comment to delete

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp [hunk 41]
```diff
@@ -0,0 +1,24 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
+** UpdatePosition
+*/
+
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp:7]: to delete

File: Sources/Server/Systems/UpdateTime/UpdateTime.cpp [hunk 42]
```diff
@@ -1,10 +1,13 @@
...
 
-void UpdateTime::operator()(Registry &r, SparseArray<components::myTime_t> &times) {
+void rtype::UpdateTime::operator()(Registry &r, SparseArray<components::myTime_s> &times) {
     (void)r;
     static std::chrono::steady_clock::time_point lastTick = std::chrono::steady_clock::now();
     std::chrono::steady_clock::time_point currentTick = std::chrono::steady_clock::now();
-    std::optional<components::myTime_t> &time = times[rtype::constants::RESERVED_ID::TIME_UPDATE];
+    // for (auto &&[time] : Zipper(times)) {
+    //     time.deltaTime = (currentTick - lastTick);
+    // }
```
Comment (Alban-Bodenes) [inline Sources/Server/Systems/UpdateTime/UpdateTime.cpp:9]: Comment

File: Sources/Client/Systems/UpdateTime/UpdateTime.cpp [hunk 43]
```diff
@@ -0,0 +1,15 @@
+#include "UpdateTime.hpp"
+
+void rtype::UpdateTime::operator()(Registry &r, SparseArray<components::myTime_s> &times) {
+    (void)r;
+    static std::chrono::steady_clock::time_point lastTick = std::chrono::steady_clock::now();
+    std::chrono::steady_clock::time_point currentTick = std::chrono::steady_clock::now();
+    // for (auto &&[time] : Zipper(times)) {
+    //     time.deltaTime = (currentTick - lastTick);
+    // }
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateTime/UpdateTime.cpp:9]: to delete

File: Sources/Client/RtypeClient.cpp [hunk 44]
```diff
@@ -1,75 +1,152 @@
...
+        false,
+        false,
+        false,
+        false,
+    };
+    _r.addComponent<components::keyState_s>(_r.entityFromIndex(rtype::constants::RESERVED_ID::GRAPH_UPDATE), std::move(key));
 }
 
 /*
  * Systems
```
Comment (Jsavang) [inline Sources/Client/RtypeClient.cpp:111]: comment to delete

File: Sources/Server/Systems/UpdateTime/UpdateTime.hpp [hunk 45]
```diff
@@ -0,0 +1,43 @@
...
+ * \brief file where the update time system is defined
+ */
+
+#ifndef UPDATETIME_HPP_
+    #define UPDATETIME_HPP_
+
+    #include "Registry.hpp"
+    #include <chrono>
+    #include "Constants.hpp"
+    //#include "Zipper.hpp"
```
Comment (Alban-Bodenes) [inline Sources/Server/Systems/UpdateTime/UpdateTime.hpp:13]: Comment

File: Sources/Client/Systems/UpdateTime/UpdateTime.cpp [hunk 46]
```diff
@@ -0,0 +1,15 @@
...
+    std::optional<components::myTime_s> &time = times[rtype::constants::RESERVED_ID::TIME_UPDATE];
+    if (time) {
+        time.value().deltaTime = (currentTick - lastTick);
+    }
```
Comment (MaelTeyssedre) [inline Sources/Client/Systems/UpdateTime/UpdateTime.cpp:13]: useless brackets

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp [hunk 47]
```diff
@@ -0,0 +1,58 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
+** File description:
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp]: epitech header to delete

**Review by MaelTeyssedre — CHANGES_REQUESTED** [review]

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp [hunk 48]
```diff
@@ -0,0 +1,23 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp:3]: epitech header to delete

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp (hunk 33 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:32]: comment to delete

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp (hunk 34 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:39]: comment to delete

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp (hunk 35 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:46]: comment to delete

File: Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp (hunk 38 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp:130]: newline to add

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp [hunk 49]
```diff
@@ -0,0 +1,22 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp:3]: epitech header to delete

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp [hunk 50]
```diff
@@ -0,0 +1,24 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp:3]: epitech header to delete

File: Sources/Client/Systems/UpdateTime/UpdateTime.cpp [hunk 51]
```diff
@@ -0,0 +1,15 @@
+#include "UpdateTime.hpp"
+
+void rtype::UpdateTime::operator()(Registry &r, SparseArray<components::myTime_s> &times) {
+    (void)r;
+    static std::chrono::steady_clock::time_point lastTick = std::chrono::steady_clock::now();
+    std::chrono::steady_clock::time_point currentTick = std::chrono::steady_clock::now();
+    // for (auto &&[time] : Zipper(times)) {
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdateTime/UpdateTime.cpp:7]: comment to delete

File: Sources/Client/Systems/Systems.hpp [hunk 52]
```diff
@@ -0,0 +1,17 @@
+/*
+** EPITECH PROJECT, 2021
+** R-TYPE
```
Comment (Jsavang) [inline Sources/Client/Systems/Systems.hpp]: header

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp [hunk 53]
```diff
@@ -0,0 +1,58 @@
+/*
+** EPITECH PROJECT, 2021
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.cpp]: header

File: Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp (hunk 48 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateDirection/UpdateDirection.hpp:3]: header

File: Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp [hunk 54]
```diff
@@ -0,0 +1,66 @@
+
+#ifndef UPDATEGRAPH_HPP_
```
Comment (Jsavang) [inline Sources/Client/Systems/UpdateGraph/UpdateGraph.hpp:2]: comment to delete

File: Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp (hunk 38 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateNetwork/UpdateNetwork.cpp:130]: newline to add

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp (hunk 49 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.cpp:3]: header

File: Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp (hunk 50 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdatePosition/UpdatePosition.hpp:3]: header

File: Sources/Client/Systems/UpdateTime/UpdateTime.cpp (hunk 51 above)
Comment (Jsavang) [inline Sources/Client/Systems/UpdateTime/UpdateTime.cpp:7]: comment to delete

File: Sources/Commons/INetwork/IPacket.hpp [hunk 55]
```diff
@@ -49,7 +49,9 @@
          * 
          * \return data deserialised
```
Comment (Jsavang) [inline Sources/Commons/INetwork/IPacket.hpp:50]: wrong doc

File: Sources/Commons/INetwork/ITCPServer.hpp [hunk 56]
```diff
@@ -63,6 +65,19 @@
              * \param client to eject
              */
             virtual void eject(size_t client) = 0;
+
+
+            // /**
+            //  * \fn std::shared_ptr<Buffer> getData()
+            //  *
```
Comment (Jsavang) [inline Sources/Commons/INetwork/ITCPServer.hpp:72]: wrong doc

File: Sources/Server/Component/HealPoint.hpp [hunk 57]
```diff
@@ -7,22 +7,24 @@
...
-     * \namespace rtype
-     * 
-     * \brief global namespace for rtype project
-     * 
-     */
     namespace rtype {
         /**
-         * \struct healPoint_s
+         * \namespace rtype
          *
```
Comment (Jsavang) [inline Sources/Server/Component/HealPoint.hpp:13]: wrong doc

File: Sources/Server/Component/Velocity.hpp [hunk 58]
```diff
@@ -7,23 +7,25 @@
...
 
-    /**
-     * \namespace rtype
-     * 
-     * \brief global namespace for rtype project
-     * 
-     */
     namespace rtype {
         /**
-         * \struct velocity_s
```
Comment (Jsavang) [inline Sources/Server/Component/Velocity.hpp:18]: wrong doc

File: Sources/Server/Component/Weapon.hpp [hunk 59]
```diff
@@ -7,22 +7,24 @@
...
-    /**
-     * \namespace rtype
-     * 
-     * \brief global namespace for rtype project
-     * 
-     */
     namespace rtype {
         /**
-         * \struct weapon_s
+         * \namespace rtype
```
Comment (Jsavang) [inline Sources/Server/Component/Weapon.hpp:12]: wrong doc

File: Sources/Server/Libs/AMonster.hpp [hunk 60]
```diff
@@ -15,169 +15,171 @@
...
+                 * \brief Set the Position object
+                 * 
+                 * \param pos position to set
+                 */
+                virtual void setPosition(std::pair<int, int> pos);
+                
+                /**
+                 * \fn virtual rtype::position_s getPosition() const
+                 * 
+                 * \brief Get the Position object
```
Comment (Jsavang) [inline Sources/Server/Libs/AMonster.hpp:146]: wrong doc

File: Sources/Client/GraphicalLib/GraphicalLib.hpp [hunk 61]
```diff
@@ -20,40 +20,83 @@
                 explicit GraphicalLib();
                 virtual ~GraphicalLib() = default;
 
-                void draw() override;
-                void play(Sound &sound) override;
-                void play(Music &music) override;
+            public:
```
Comment (Jsavang) [inline Sources/Client/GraphicalLib/GraphicalLib.hpp:23]: comment to delete

File: Sources/Client/GraphicalLib/IGraphicalLib.hpp (hunk 4 above)
Comment (Jsavang) [inline Sources/Client/GraphicalLib/IGraphicalLib.hpp:33]: comment to delete

File: Sources/Client/Network/TCP/cTCPClient.hpp [hunk 62]
```diff
@@ -15,100 +15,106 @@
...
+         * \class TCPClient TCPCLient.hpp
+         * 
+         * \brief class that contain methode to manage tcp client 
+         * 
+         */
+        class TCPClient : public ITCPClient {
+            public:
+                /**
+                 * \fn explicit TCPCLient()
+                 *
```
Comment (Jsavang) [inline Sources/Client/Network/TCP/cTCPClient.hpp:29]: wrong doc

File: Sources/Client/Network/TCP/cTCPClient.hpp [hunk 63]
```diff
@@ -15,100 +15,106 @@
...
+                 * 
+                 * \brief ctor TCPClient
+                 * 
+                 */
+                explicit TCPClient() = default;
+
+                /**
+                 * \brief Construct a new TCPClient object
+                 * 
+                 * \param context asio context
```
Comment (Jsavang) [inline Sources/Client/Network/TCP/cTCPClient.hpp:38]: wrong doc

**Review by Jsavang — CHANGES_REQUESTED** [review]

**Review by Jsavang — APPROVED** [review]

**Review by MaelTeyssedre — APPROVED** [review]

**Pull request merged**